### ISSUE TITLE
[ENH] Generalize training options for pre-trained models

### ISSUE THREAD

**Issue by geetu040** [sktime/sktime#7426]
since we have a bunch of pre-trained models in sktime, we now want to make them more coherent and comparable functionality especially with training strategy.

I have collected the implemented training strategies for the following models
`HFPatchTSTForecaster`, `HFPatchTSTForecaster`,`TimesFMForecaster`, `TinyTimeMixerForecaster`

Based on how the models weights were initialized, this depends on the configuration, we can have following options, for each option we have the corresponding training strategies

| Model Initialization / Fit Strategy        | Zero-Shot (zero-shot)                                                                                                                          | PEFT  (peft)                                      | Train All Weights (full)                          | Train Only Random Weights (minimal)                                                                                   |
| ------------------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------- | ------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------- |
| **Random Weights**                         | Should Raise Error                                                                                                                             | Should Raise Error                                | same as "minimal"                                 | `HFPatchTSTForecaster.mode == "untrained"`                                                                            |
| **Pre-trained Weights**                    | `HFPatchTSTForecaster.mode == "zero-shot"`, `TimesFMForecaster` (only option), `TinyTimeMixerForecaster` (selected automatically for this row) | `HFTransformersForecaster.fit_strategy == "peft"` | `HFPatchTSTForecaster.mode == "finetune"`         | nothing happens here                                                                                                  |
| **Some Pre-trained & Some Random Weights** | Should Raise Error                                                                                                                             | Should Raise Error                                | `HFTransformersForecaster.fit_strategy == "full"` | `HFTransformersForecaster.fit_strategy == "minimal"`, `TinyTimeMixerForecaster` (selected automatically for this row) |

FYI: @julian-fong @benHeid @pranavvp16 @XinyuWuu

**Comment by julian-fong** [comment]
Would an extension template be helpful as a reference for new global forecasters that contributors want to implement?

**Comment by geetu040** [comment]
> Would an extension template be helpful as a reference for new global forecasters that contributors want to implement?

yes that sounds good, that template should also cover the caching behavior as is implemented with `TimesFM` and `Chronos`

**Comment by Sohaib-Ahmed21** [comment]
@fkiraly @geetu040
I want to work on this issue, can I get some pointers on how to approach it?

**Comment by geetu040** [comment]
This issue basically collects how the training options for different pre-trained models work and how we can keep them more coherent
Is there a particular item or feature you have in mind to work on related to this?

**Comment by Sohaib-Ahmed21** [comment]
> This issue basically collects how the training options for different pre-trained models work and how we can keep them more coherent Is there a particular item or feature you have in mind to work on related to this?

`TinyTimeMixer ` currently automatically adjusts which **fit_strategy** to use, we can take fit_strategy from user. I particularly want to work on it.

**Comment by geetu040** [comment]
> `TinyTimeMixer ` currently automatically adjusts which **fit_strategy** to use, we can take fit_strategy from user. I particularly want to work on it.

yes, I think it would be great to have `fit_strategy` option for `TinyTimeMixer`. We have discussed this on discord before.

FYI: @fkiraly @benHeid 

`TinyTimeMixer` currently doesnot take a `fit_strategy` from user. It trains only the weights that have been newly initialized because of a change in configuration, rest of the weights are always frozen, this mimics the `"minimal"` mode of `fit_strategy` compared to other pre-trained models. Do we want to give `fit_strategy` option to the user as another model parameter? This does not break the API and ensures backward compatibility.

**Comment by geetu040** [comment]
@Sohaib-Ahmed21 @julian-fong 
what do you guys think about dropping the option of "zero-shot" from `fit_strategy` from the design, because if `fit_strategy=="minimal"` and config is unchanged, it automatically performs "zero-shot", we can inform this through the docs. This is exactly how `HFTransformersForecaster` works.
I think adding the option for "zero-shot" is just going to add more code and validation for the same features.

FYI: @benHeid @fkiraly

**Comment by julian-fong** [comment]
I think the nice thing about having the "zero-shot" option is to ease up understanding of the modes. It makes it very obvious that the option is designed to allow the user to pass in pretrained model to do "zero-shot" forecasting. If we pass it into minimal, it may confuse users that don't bother to comb through every word of the docstring. But I'm fine with either way

Also, I don't think most users will be trying to pass in configs to do zero-shot, wouldn't the most standard case would be to pass in their own `model_path`?

**Comment by Sohaib-Ahmed21** [comment]
> I think the nice thing about having the "zero-shot" option is to ease up understanding of the modes. It makes it very obvious that the option is designed to allow the user to pass in pretrained model to do "zero-shot" forecasting. If we pass it into minimal, it may confuse users that don't bother to comb through every word of the docstring. But I'm fine with either way

I agree with you. Thoughts from core-devs will be helpful. @fkiraly @geetu040 @benHeid 
> Also, I don't think most users will be trying to pass in configs to do zero-shot, wouldn't the most standard case would be to pass in their own `model_path`?

Yes, and I think we're currently right in this perspective, allowing users to pass in their own model_path with default config being used.

**Comment by geetu040** [comment]
@fkiraly @benHeid do you have a preference with the "zero-shot" as discussed here https://github.com/sktime/sktime/issues/7426#issuecomment-2598842657, your input will be really appreciated, https://github.com/sktime/sktime/pull/6787 and https://github.com/sktime/sktime/pull/7648 are blocked on this decision

**Comment by geetu040** [comment]
@benHeid from discord
> Mhm... I think each way is fine at least if it is documented. My current favourite would be 
> - minimal: Fit everything that needs to be fitted (might also be fit nothing)
> - zero-shot: Check if not fit is needed, otherwise raise an exception
> 
> So I think the overlap is fine, it is only important that it is well documented
